src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerLimits.cs

```diff
@@ -23,8 +23,10 @@ public class KestrelServerLimits
         // Matches the default LimitRequestFields in Apache httpd.
         private int _maxRequestHeaderCount = 100;
 
-        // Matches the default http.sys connection timeout.
-        private TimeSpan _connectionTimeout = TimeSpan.FromMinutes(2);
+        // Matches the default http.sys connectionTimeout.
+        private TimeSpan _keepAliveTimeout = TimeSpan.FromMinutes(2);
+
+        private TimeSpan _requestHeadersTimeout = TimeSpan.FromSeconds(30);
```

Comment:
It's the midpoint for the header timeout part of Apache's `RequestReadTimeout`. Their default is 20-40 seconds (it increases if data is coming at a minimum rate). nginx uses 60s.